src/main/java/cn/nukkit/entity/Entity.java

```diff
@@ -1827,7 +1828,20 @@ public boolean isOnLadder() {
 
         return b.getId() == Block.LADDER;
     }
-
+    public boolean applyNametag(Player player, Item item)
+    {
+        if(item.hasCustomName())
+        {
+            this.setNameTag(item.getCustomName());
+            this.setNameTagVisible(true);
+            if(player.isSurvival()) { player.getInventory().removeItem(item); }
```

Comment:
I know that the original code used `removeItem`, I didn't actually test yet so I may be wrong but I think it would remove the entire stack, so if a player has 64 name tags in his hand, all them would be consumed. Maybe a `item.setCount(item.getCount() -1)` would be better.

Reply:
Fixed on the next commit.

Reply:
I've made some improvements, I removed the direct item manipulation because the server already does that when `onInteract` returns `true`